geoutils/georaster.py

```diff
@@ -822,10 +913,179 @@ def format_value(value):
             raise ValueError(
                 'Value provided for band was not int or None.')
 
-        if return_window == True:
+        if return_window:
             return (value, win)
         else:
             return value
-    
+
+    def xxyy(self, offset='corner', grid=True):
+        """
+        Get x,y coordinates of all pixels in the raster.
+
+        :param offset: coordinate type. If 'corner', returns corner coordinates of pixels.
+            If 'center', returns center coordinates. Default is corner.
+        :param grid: Return gridded coordinates. Default is True.
+        :type offset: str
+        :type grid: bool
+        :returns x,y: numpy arrays corresponding to the x,y coordinates of each pixel.
+        """
+        assert offset in ['corner', 'center'], "ctype is not one of 'corner', 'center': {}".format(offset)
+
+        xmin, ymin, xmax, ymax = self.ds.bounds
+        dx = list(self.ds.transform)[0]
+        dy = list(self.ds.transform)[4]
+        npix_x = self.ds.width
+        npix_y = self.ds.height
+
+        # transform = Affine.from_gdal(self.ds.transform)
+        # nx, ny = self.ds.width, self.ds.height
+        # xx, yy = np.meshgrid(np.arange(nx) + 0.5, np.arange(ny) + 0.5) * transform
+
+        # return xx, yy
+
+        if dx < 0:
```

Comment:
You could do this in one line
`xx = np.linspace(xmin, xmax, npix_x + 1)[::np.sign(dx)]`
and same for yy.